### PR TITLE
App install on Heroku

### LINKED ISSUE #337: Heroku install?
maybe we should have a simple heroku install that people can get up and running with

### PR DESCRIPTION
**Pull request by AkshJain99**
Here's link for the app https://rocky-inlet-65308.herokuapp.com

fixes #337

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -17,3 +17,4 @@ six==1.11.0
 ujson==1.35
 yarl==1.2.6
 zope.interface==4.4.3
+gunicorn==19.9.0
```
Comment (vangheem) [inline requirements.txt]: let's not depend on gunicorn here.

If it's just for heroku install, add a different requirements file or just include the version number in that definition directly.

That being said, why is gunicorn is necessary?

Comment (AkshJain99) [reply requirements.txt]: @vangheem also if i am not creating procfile and not adding gunicorn version in requirement.txt i am getting this error
```
akshit@akshit-Inspiron-5567:~/guillotina$ heroku logs --tail
2019-01-20T20:40:32.513556+00:00 app[web.1]: [2019-01-20 20:40:32 +0000] [10] [INFO] Worker exiting (pid: 10)
2019-01-20T20:40:33.067114+00:00 heroku[web.1]: State changed from up to crashed
2019-01-20T20:40:33.042166+00:00 heroku[web.1]: Process exited with status 1
2019-01-20T20:40:39.546004+00:00 app[web.1]: [2019-01-20 20:40:39 +0000] [4] [INFO] Starting gunicorn 19.9.0
2019-01-20T20:40:39.546496+00:00 app[web.1]: [2019-01-20 20:40:39 +0000] [4] [INFO] Listening at: http://0.0.0.0:9863 (4)
2019-01-20T20:40:39.546614+00:00 app[web.1]: [2019-01-20 20:40:39 +0000] [4] [INFO] Using worker: sync
2019-01-20T20:40:39.550132+00:00 app[web.1]: [2019-01-20 20:40:39 +0000] [10] [INFO] Booting worker with pid: 10
2019-01-20T20:40:39.638217+00:00 app[web.1]: [2019-01-20 20:40:39 +0000] [12] [INFO] Booting worker with pid: 12
2019-01-20T20:40:39.741141+00:00 app[web.1]: [2019-01-20 20:40:39 +0000] [10] [ERROR] Exception in worker process
2019-01-20T20:40:39.741145+00:00 app[web.1]: Traceback (most recent call last):
2019-01-20T20:40:39.741147+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/arbiter.py", line 583, in spawn_worker
2019-01-20T20:40:39.741149+00:00 app[web.1]: worker.init_process()
2019-01-20T20:40:39.741151+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/workers/base.py", line 129, in init_process
2019-01-20T20:40:39.741153+00:00 app[web.1]: self.load_wsgi()
2019-01-20T20:40:39.741154+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/workers/base.py", line 138, in load_wsgi
2019-01-20T20:40:39.741156+00:00 app[web.1]: self.wsgi = self.app.wsgi()
2019-01-20T20:40:39.741158+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/app/base.py", line 67, in wsgi
2019-01-20T20:40:39.741159+00:00 app[web.1]: self.callable = self.load()
2019-01-20T20:40:39.741161+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 52, in load
2019-01-20T20:40:39.741162+00:00 app[web.1]: return self.load_wsgiapp()
2019-01-20T20:40:39.741164+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 41, in load_wsgiapp
2019-01-20T20:40:39.741166+00:00 app[web.1]: return util.import_app(self.app_uri)
2019-01-20T20:40:39.741168+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/util.py", line 350, in import_app
2019-01-20T20:40:39.741169+00:00 app[web.1]: __import__(module)
2019-01-20T20:40:39.741171+00:00 app[web.1]: File "/app/guillotina/__init__.py", line 14, in <module>
2019-01-20T20:40:39.741172+00:00 app[web.1]: __version__ = pkg_resources.get_distribution('guillotina').version
2019-01-20T20:40:39.741174+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/pkg_resources/__init__.py", line 472, in get_distribution
2019-01-20T20:40:39.741176+00:00 app[web.1]: dist = get_provider(dist)
2019-01-20T20:40:39.741177+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/pkg_resources/__init__.py", line 344, in get_provider
2019-01-20T20:40:39.741179+00:00 app[web.1]: return working_set.find(moduleOrReq) or require(str(moduleOrReq))[0]
2019-01-20T20:40:39.741180+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/pkg_resources/__init__.py", line 892, in require
2019-01-20T20:40:39.741182+00:00 app[web.1]: needed = self.resolve(parse_requirements(requirements))
2019-01-20T20:40:39.741184+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/pkg_resources/__init__.py", line 778, in resolve
2019-01-20T20:40:39.741185+00:00 app[web.1]: raise DistributionNotFound(req, requirers)
2019-01-20T20:40:39.741187+00:00 app[web.1]: pkg_resources.DistributionNotFound: The 'guillotina' distribution was not found and is required by the application
2019-01-20T20:40:39.741701+00:00 app[web.1]: [2019-01-20 20:40:39 +0000] [10] [INFO] Worker exiting (pid: 10)
2019-01-20T20:40:39.808058+00:00 app[web.1]: [2019-01-20 20:40:39 +0000] [12] [ERROR] Exception in worker process
2019-01-20T20:40:39.808077+00:00 app[web.1]: Traceback (most recent call last):
2019-01-20T20:40:39.808083+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/arbiter.py", line 583, in spawn_worker
2019-01-20T20:40:39.808085+00:00 app[web.1]: worker.init_process()
2019-01-20T20:40:39.808087+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/workers/base.py", line 129, in init_process
2019-01-20T20:40:39.808088+00:00 app[web.1]: self.load_wsgi()
2019-01-20T20:40:39.808090+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/workers/base.py", line 138, in load_wsgi
2019-01-20T20:40:39.808091+00:00 app[web.1]: self.wsgi = self.app.wsgi()
2019-01-20T20:40:39.808093+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/app/base.py", line 67, in wsgi
2019-01-20T20:40:39.808094+00:00 app[web.1]: self.callable = self.load()
2019-01-20T20:40:39.808096+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 52, in load
2019-01-20T20:40:39.808097+00:00 app[web.1]: return self.load_wsgiapp()
2019-01-20T20:40:39.808099+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 41, in load_wsgiapp
2019-01-20T20:40:39.808100+00:00 app[web.1]: return util.import_app(self.app_uri)
2019-01-20T20:40:39.808102+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/util.py", line 350, in import_app
2019-01-20T20:40:39.808103+00:00 app[web.1]: __import__(module)
2019-01-20T20:40:39.808109+00:00 app[web.1]: File "/app/guillotina/__init__.py", line 14, in <module>
2019-01-20T20:40:39.808110+00:00 app[web.1]: __version__ = pkg_resources.get_distribution('guillotina').version
2019-01-20T20:40:39.808118+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/pkg_resources/__init__.py", line 472, in get_distribution
2019-01-20T20:40:39.808120+00:00 app[web.1]: dist = get_provider(dist)
2019-01-20T20:40:39.808121+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/pkg_resources/__init__.py", line 344, in get_provider
2019-01-20T20:40:39.808122+00:00 app[web.1]: return working_set.find(moduleOrReq) or require(str(moduleOrReq))[0]
2019-01-20T20:40:39.808124+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/pkg_resources/__init__.py", line 892, in require
2019-01-20T20:40:39.808125+00:00 app[web.1]: needed = self.resolve(parse_requirements(requirements))
2019-01-20T20:40:39.808127+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/pkg_resources/__init__.py", line 778, in resolve
2019-01-20T20:40:39.808128+00:00 app[web.1]: raise DistributionNotFound(req, requirers)
2019-01-20T20:40:39.808130+00:00 app[web.1]: pkg_resources.DistributionNotFound: The 'guillotina' distribution was not found and is required by the application
2019-01-20T20:40:39.808968+00:00 app[web.1]: [2019-01-20 20:40:39 +0000] [12] [INFO] Worker exiting (pid: 12)
2019-01-20T20:40:39.857208+00:00 app[web.1]: Traceback (most recent call last):
2019-01-20T20:40:39.857215+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/arbiter.py", line 210, in run
2019-01-20T20:40:39.857659+00:00 app[web.1]: self.sleep()
2019-01-20T20:40:39.857664+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/arbiter.py", line 360, in sleep
2019-01-20T20:40:39.857978+00:00 app[web.1]: ready = select.select([self.PIPE[0]], [], [], 1.0)
2019-01-20T20:40:39.857999+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/arbiter.py", line 245, in handle_chld
2019-01-20T20:40:39.858246+00:00 app[web.1]: self.reap_workers()
2019-01-20T20:40:39.858268+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/arbiter.py", line 525, in reap_workers
2019-01-20T20:40:39.858642+00:00 app[web.1]: raise HaltServer(reason, self.WORKER_BOOT_ERROR)
2019-01-20T20:40:39.858693+00:00 app[web.1]: gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>
2019-01-20T20:40:39.858697+00:00 app[web.1]: 
2019-01-20T20:40:39.858698+00:00 app[web.1]: During handling of the above exception, another exception occurred:
2019-01-20T20:40:39.858699+00:00 app[web.1]: 
2019-01-20T20:40:39.858720+00:00 app[web.1]: Traceback (most recent call last):
2019-01-20T20:40:39.858739+00:00 app[web.1]: File "/app/.heroku/python/bin/gunicorn", line 11, in <module>
2019-01-20T20:40:39.858893+00:00 app[web.1]: sys.exit(run())
2019-01-20T20:40:39.858898+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 61, in run
2019-01-20T20:40:39.859013+00:00 app[web.1]: WSGIApplication("%(prog)s [OPTIONS] [APP_MODULE]").run()
2019-01-20T20:40:39.859016+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/app/base.py", line 223, in run
2019-01-20T20:40:39.859232+00:00 app[web.1]: super(Application, self).run()
2019-01-20T20:40:39.859234+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/app/base.py", line 72, in run
2019-01-20T20:40:39.859368+00:00 app[web.1]: Arbiter(self).run()
2019-01-20T20:40:39.859369+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/arbiter.py", line 232, in run
2019-01-20T20:40:39.860051+00:00 app[web.1]: self.halt(reason=inst.reason, exit_status=inst.exit_status)
2019-01-20T20:40:39.860055+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/arbiter.py", line 345, in halt
2019-01-20T20:40:39.861067+00:00 app[web.1]: self.stop()
2019-01-20T20:40:39.861087+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/arbiter.py", line 393, in stop
2019-01-20T20:40:39.861422+00:00 app[web.1]: time.sleep(0.1)
2019-01-20T20:40:39.861423+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/arbiter.py", line 245, in handle_chld
2019-01-20T20:40:39.861751+00:00 app[web.1]: self.reap_workers()
2019-01-20T20:40:39.861771+00:00 app[web.1]: File "/app/.heroku/python/lib/python3.6/site-packages/gunicorn/arbiter.py", line 525, in reap_workers
2019-01-20T20:40:39.862137+00:00 app[web.1]: raise HaltServer(reason, self.WORKER_BOOT_ERROR)
2019-01-20T20:40:39.862167+00:00 app[web.1]: gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>
2019-01-20T20:40:55.582721+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=guillotina.herokuapp.com request_id=5d88f318-3b10-45e8-873c-803f6c37bf7e fwd="14.139.240.82" dyno= connect= service= status=503 bytes= protocol=https
2019-01-20T20:40:55.866313+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/favicon.ico" host=guillotina.herokuapp.com request_id=3dce1291-81f5-4dd1-95cc-761b9b69b5b8 fwd="14.139.240.82" dyno= connect= service= status=503 bytes= protocol=https
2019-01-20T20:51:49.122743+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=guillotina.herokuapp.com request_id=8c35af6b-55a5-49d1-921b-bacba01ed5fc fwd="14.139.240.82" dyno= connect= service= status=503 bytes= protocol=https
2019-01-20T20:53:17.436326+00:00 heroku[router]: at=error code=H10 desc="App crashed" method=GET path="/" host=guillotina.herokuapp.com request_id=691f68da-84ba-4854-b34e-77d08412293c fwd="14.139.240.82" dyno= connect= service= status=503 bytes= protocol=https
```

**Comment by vangheem** [comment]
Is this still being worked on?

I'm closing for now. Please reopen if it should still be open or when there is more progress..

Also, please add "WIP" to PR if it's not working/ready to merge.

**Comment by AkshJain99** [comment]
WIP

**Pull request closed without merging**